cairo/ethereum/prague/fork.cairo

```diff
@@ -1124,6 +1129,20 @@ func check_transaction{
     let range_check_ptr = [ap - 2];
     let max_gas_fee = Uint([ap - 1]);
 
+    let is_not_blob_or_set_code_transaction = (TransactionType.BLOB - transaction_type) * (
+        TransactionType.SET_CODE - transaction_type
+    );
+    let to = get_to(tx);
+    if (is_not_blob_or_set_code_transaction == FALSE and cast(to.value, felt) == 0) {
+        raise('InvalidBlock');
+    }
+
+    let is_not_set_code_transaction = TransactionType.SET_CODE - transaction_type;
+    let authorizations_len = get_authorizations_len_unchecked(tx);
```

Comment:
cannot use `authorizations.value.len`?

Reply:
If not a SetCodeTx this will raise.
So a special function is needed to avoid doing imbricated if clutter

Reply:
not sure I get it, `authorizations` you get from `authorizations` which returns an object with `len:=0` if it's not a SetCodeTransaction